semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java

```diff
@@ -474,13 +484,23 @@ public static URL getNormalizedUrl(String url) throws MalformedURLException {
   public static Optional<Integer> getHttpRequestSize(Event event) {
     String httpRequestSize =
         SpanAttributeUtils.getFirstAvailableStringAttribute(event, REQUEST_SIZE_ATTRIBUTES);
-    return Optional.ofNullable(httpRequestSize).map(Integer::parseInt);
+
+    Optional<String> requestSize = Optional.ofNullable(httpRequestSize);
+    if (!requestSize.isEmpty()) return requestSize.map(Integer::parseInt);
+
+    String requestBody = SpanAttributeUtils.getStringAttribute(event, HTTP_REQUEST_BODY);
```

Comment:
It's a corner case, but we should also check to see whether the body has been truncated.  If it has been truncated, I think we should just set the size to 0, as we don't know what the actual size was.

Reply:
@davexroth / @avinashkolluru How do we know if the body is truncated? Is there any attribute that indicated that?